src/hipoc/hipoc_kernel.cpp

```diff
@@ -56,7 +58,11 @@ void HIPOCKernelInvoke::run(void* args, std::size_t size) const
         stop  = make_hip_event();
     }
 
-    // std::cerr << "Launch kernel: " << name << std::endl;
+    const char* const arch = miopen::GetStringEnv(MIOPEN_DEVICE_ARCH{});
+    if(arch != nullptr && strlen(arch) > 0)
+    {
+        MIOPEN_THROW("Escaping launching kernel");
```

Comment:
Not an error when arch matches machine. More descriptive error would be better. eg compile arch does not match, escaping launching kernel / MIOPEN_DEVICE_ARCH was used, escaping launching kernel.

Reply:
the idea here is that if we use that special env var at all then we are compiling and running separately so we want to escape the rest of the execution. This is why i don't check if the env var matches the underlying arch.
But i can make the error more descriptive.